File: beginner/days-left-in-year.py
import datetime

def days_left_in_year(date=None):
    if date is None:
        today = datetime.date.today()
    else:
        today = datetime.datetime.strptime(date, '%Y-%m-%d').date()

    end_of_year = datetime.date(today.year, 12, 31)
    days_left = (end_of_year - today).days
    return days_left

# Ask user for input
while True:
    user_input = input("Enter a specific date (YYYY-MM-DD) or press Enter for today's date: ")
    
    try:
        if user_input:
            days_remaining = days_left_in_year(date=user_input)
        else:
            days_remaining = days_left_in_year()

        # Output the result
        print(f"There are {days_remaining} days left in the year.")
        break

    except ValueError:
        print("Invalid date format. Please enter the date in YYYY-MM-DD format.")

File: beginner/days-left-in-year.js
const readline = require("readline-sync");
//* npm install readline-sync
const moment = require("moment");
//* npm install moment

//? Ask user for input
while (true) {
  let userInput = readline.question(
    "Enter a specific date (YYYY-MM-DD) or press Enter for today's date: "
  );

  try {
    let daysRemaining = daysLeftInYear(userInput);

    // Output the result
    console.log(`There are ${daysRemaining} days left in the year.`);
    break;
  } catch (error) {
    console.log("Invalid date format. Please enter the date in YYYY-MM-DD format.");
  }
}

function daysLeftInYear(date) {
  let today;
  if (date) {
    today = moment(date, "YYYY-MM-DD");
  } else {
    today = moment();
  }

  let endOfYear = moment(today).endOf("year");
  let daysLeft = endOfYear.diff(today, "days");
  return daysLeft;
}